plugins/schedule/base/lib/model/ScheduleEvent.php

```diff
@@ -270,15 +270,29 @@ public function getTemplateEntryCategoriesIdsForIndex()
 		return '';
 	}
 
-	public function getSystemNamesForIndex()
+	public function getTemplateEntryIdForIndex()
+	{
+		return '';
+	}
+
+
+	public function getResourceSystemNamesForIndex()
 	{
 		$eventResources = ScheduleEventResourcePeer::retrieveByEventId($this->getId());
 
 		$system_names = array();
 		foreach($eventResources as $eventResource)
 		{
 			$resource = ScheduleResourcePeer::retrieveByPK($eventResource->getResourceId());

```

Comment:
Please build array of all resource ids and access the DB only once with that list instead of accessing the DB once for each id.